### PR TITLE
feat(search): Redesign search page - left filter pane

### PR DESCRIPTION
**Pull request by dexter-mh-lee**
TL;DR Redesign all entities search page to be a single list of search results with a left filter pane.
- As opposed to the current page with one module per entity type and tabs to navigate between different types

This makes navigation after search easier with clear filters users can apply from the get go. 

To make this happen, we made the following changes
1) Add searchAcrossEntities end point which aggregates search results from multiple EntitySearchService (one per entity - pervious SearchService)
  - AllEntitiesSearchAggregator (responsible for doing aggregations above) fetches results in parallel and combines the filters returned by each EntitySearchService
2) Add SearchRanker which can rank entities returned by search based on a set of features: each search ranker defines a set of feature extractors and logic to generate a comparable score from the features
3) Refactor GraphQL resolvers for search and searchAcrossEntities - no need to use EntityType class to figure out how to call search
4) Frontend renders filters in a useful way: not just string values, but if entity, render the entity - like tags, glossary terms, platforms

** Breaking Changes **
1) Code was previously using Search response related PDL objects (like SearchResult, SearchResultMetadata) from datahub-gms (package com.linkedin.metadata.query). Changed it to use objects in datahub (package com.linkedin.metadata.search)
2) SearchResult now returns a list of SearchEntity. Each SearchEntity contains the urn of the result entity and metadata about them (like highlights, features, etc.) - Note, previously highlights were inside SearchResultMetadata, and we had to zip it with the search result list to figure out which highlights to show for each result

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

**Comment by jjoyce0510** [comment]
A ton of work here. Really excited to see how the community receives it. Seems that some test is failing, though.
